datafusion/sql/src/parser.rs

```diff
@@ -1009,34 +996,33 @@ mod tests {
             columns: vec![make_column_def("c1", DataType::Int(None))],
             file_type: "CSV".to_string(),
             has_header: false,
-            delimiter: ',',
             location: "foo.csv".into(),
             table_partition_cols: vec![],
             order_exprs: vec![],
             if_not_exists: false,
-            file_compression_type: UNCOMPRESSED,
             unbounded: false,
             options: HashMap::new(),
             constraints: vec![],
         });
         expect_parse_ok(sql, expected)?;
 
         // positive case with delimiter
-        let sql = "CREATE EXTERNAL TABLE t(c1 int) STORED AS CSV DELIMITER '|' LOCATION 'foo.csv'";
+        let sql = "CREATE EXTERNAL TABLE t(c1 int) STORED AS CSV LOCATION 'foo.csv' OPTIONS ('format.delimiter' '|')";
```

Comment:
Is it required to removing the existing syntax support for `DELIMTER` and `COMPRESSION` -- I think other users will see that as a regression. 

It seems like it would be possible to apply the same type of logic for `has_header` to these other options to prevent conflicts and silent igores